bokeh/tests/test_objects.py

```diff
@@ -639,5 +642,42 @@ def mutate(x):
                              dict(a=1, b=2, c=3),
                              dict(a=1, b=7, c=8))
 
+
+class BaseTwinAxis(object):
+    """Base class for testing extra ranges"""
+
+    def verify_axis(self, axis_name):
+        plot = figure()  # no need for setUp()
+        range_obj = getattr(plot, 'extra_{}_ranges'.format(axis_name))
+        range_obj['foo_range'] = self.get_range_instance()
+        self.assertIsNone(show(plot))
```

Comment:
First, tests! This is great, a few suggestions tho. I think this might be a more appropriate place for these tests:

https://github.com/bokeh/bokeh/blob/master/bokeh/models/tests/test_plots.py

Then I'd mention `show` tries to open a browser, I am a little concerned that it might cause some issue in some testing configuration. The validation happens immediately on the set attr, so I think it can simply be omitted. Also as matter of conceptual purity, I think it would be good to avoid imports from `bokeh.plotting` in this tests under `bokeh.models`. May I suggest:
```
def verify_axis(self, axis_name):
     plot = Plot()  # no need for setUp()
     range_obj = getattr(plot, 'extra_{}_ranges'.format(axis_name))
     range_obj['foo_range'] = self.get_range_instance()
```